packages/wdio-config/src/utils.ts

```diff
@@ -162,31 +167,33 @@ export function validateConfig (defaults, options, keysToKeep = []) {
         }
 
         if (typeof options[name] === 'undefined' && expectedOption.default) {
+            // @ts-ignore Expression produces a union type that is too complex to represent for Typescript.
             params[name] = expectedOption.default
         }
 
         if (typeof options[name] !== 'undefined') {
-            if (typeof options[name] !== expectedOption.type) {
+            const optValue = options[name]
+            if (typeof optValue !== expectedOption.type) {
                 throw new Error(`Expected option "${name}" to be type of ${expectedOption.type} but was ${typeof options[name]}`)
             }
 
             if (typeof expectedOption.validate === 'function') {
                 try {
-                    expectedOption.validate(options[name])
+                    expectedOption.validate(optValue as never) // TS limitation
                 } catch (e) {
                     throw new Error(`Type check for option "${name}" failed: ${e.message}`)
                 }
             }
 
-            if (expectedOption.match && !options[name].match(expectedOption.match)) {
+            if (typeof optValue === 'string' && expectedOption.match && !optValue.match(expectedOption.match)) {
```

Comment:
```suggestion
            if (typeof optValue === 'string' && !optValue.match(expectedOption.match)) {
```

Reply:
Are you sure about this change? The behaviour won't be the same for all falsy values of `expectedOption.match`.
If `expectedOption.match` is `undefined` or `''` , `optValue.match(expectedOption.match)` will give a `truthy` value.
But if `expectedOption.match` is `null` or `false` or `0`, `optValue.match(expectedOption.match)` will give `null` value.

Reply:
you right!